src/libraries/System.Reflection.MetadataLoadContext/src/System/Reflection/TypeLoading/Modules/GetTypeCoreCache.cs

```diff
@@ -77,7 +77,7 @@ private Container(GetTypeCoreCache owner, int[] buckets, Entry[] entries, int ne
                 _owner = owner;
             }
 
-            public bool TryGetValue(ReadOnlySpan<byte> ns, ReadOnlySpan<byte> name, int hashCode, out RoDefinitionType value)
+            public bool TryGetValue(ReadOnlySpan<byte> ns, ReadOnlySpan<byte> name, int hashCode, out RoDefinitionType? value)
```

Comment:
Seems parameter `value` should have `[NotNullWhen(true)]`, then `GetOrAdd (...)` method above should return non null (it is creating new if TryGet was not successful)